### PR TITLE
Fix static method detection on Python 3.

### PR DESCRIPTION
**Pull request by mhils**
We erroneously reported all methods as static, this PR fixes this.

/cc @mitsuhiko

### CODE REVIEW

File: pdoc/doc.py [hunk 1]
```diff
@@ -82,6 +83,17 @@ def _is_exported(ident_name):
     return not ident_name.startswith("_")
 
 
+def _is_static(cls: typing.Type, method_name: str) -> bool:
+    """
+    Returns `True` if the given method is a @staticmethod.
+    """
+    for c in inspect.getmro(cls):
+        if method_name in c.__dict__:
+            return isinstance(c.__dict__[method_name], staticmethod)
```
Comment (mitsuhiko) [inline pdoc/doc.py]: I don't think this check is sufficient. We use `classmethod` and others too to get things at class level.

**Comment by mitsuhiko** [comment]
I also just realized that right now `classmethod` is reported as instance method.

**Comment by mhils** [comment]
Yes, this needs to be handled as well - the lack of any tests does hurt. Will fix this after lunch. :)

**Comment by mitsuhiko** [comment]
i wonder if resolving the descriptor would be a better solution (eg: `c.__dict__[method_name].__get__(None, c)`) and then testing what's on there.

**Comment by mhils** [comment]
> i wonder if resolving the descriptor would be a better solution (eg: `c.__dict__[method_name].__get__(None, c)`) and then testing what's on there.

That won't help to distinguish static methods and regular methods I think:
```python
>>> Dummy.__dict__["method"].__get__(None, Dummy)
<function Dummy.method at 0x03BAB300>
>>> Dummy.__dict__["class_method"].__get__(None, Dummy)
<bound method Dummy.class_method of <class '__main__.Parent'>>
>>> Dummy.__dict__["static_method"].__get__(None, Dummy)
<function Dummy.static_method at 0x03BAB390>
```

I pushed a revised version that should detect both class- and staticmethods. Thanks for the quick review earlier!

**Pull request merged**